relay-server/src/services/processor.rs

```diff
@@ -1207,96 +1207,90 @@ impl EnvelopeProcessorService {
+        else {
+            return Ok(());
+        };
+
+        if !tx_config.is_enabled() || state.project_state.has_feature(Feature::DiscardTransaction) {
+            return Ok(());
+        }
 
-            if let Some(config) = config {
-                let combined_config = CombinedMetricExtractionConfig::new(global_config, config);
+        let Some(event) = state.event.value() else {
+            return Ok(());
+        };
 
-                let metrics = crate::metrics_extraction::event::extract_metrics(
-                    event,
-                    state.spans_extracted,
-                    &combined_config,
-                    self.inner
-                        .config
-                        .aggregator_config_for(MetricNamespace::Spans)
-                        .max_tag_value_length,
-                    global.options.span_extraction_sample_rate,
-                );
-                state.event_metrics_extracted |= !metrics.is_empty();
-                state.extracted_metrics.project_metrics.extend(metrics);
-            }
+        let metrics = crate::metrics_extraction::event::extract_metrics(
+            event,
+            state.spans_extracted,
+            &generic_config,
+            self.inner
+                .config
+                .aggregator_config_for(MetricNamespace::Spans)
+                .max_tag_value_length,
+            global.options.span_extraction_sample_rate,
+        );
+        state.extracted_metrics.project_metrics.extend(metrics);
 
-            if let Some(ErrorBoundary::Ok(ref tx_config)) =
-                state.project_state.config.transaction_metrics
-            {
-                if tx_config.is_enabled()
-                    && !state.project_state.has_feature(Feature::DiscardTransaction)
-                {
-                    let transaction_from_dsc = state
-                        .managed_envelope
-                        .envelope()
-                        .dsc()
-                        .and_then(|dsc| dsc.transaction.as_deref());
-
-                    let extractor = TransactionExtractor {
-                        config: tx_config,
-                        generic_tags: config.map(|c| c.tags.as_slice()).unwrap_or_default(),
-                        transaction_from_dsc,
-                        sampling_result,
-                        has_profile: state.profile_id.is_some(),
-                    };
-
-                    state.extracted_metrics.extend(extractor.extract(event)?);
-                    state.event_metrics_extracted |= true;
-                }
-            }
+        let transaction_from_dsc = state
+            .managed_envelope
+            .envelope()
+            .dsc()
+            .and_then(|dsc| dsc.transaction.as_deref());
+
+        let extractor = TransactionExtractor {
+            config: tx_config,
+            generic_config: Some(&generic_config),
```

Comment:
This is the actual bug fix: Tag definitions are applied from both global and project config, not just project config.